### PR TITLE
Removed multiple Utf8 classes

### PR DESCRIPTION
**Pull request by KrzysztofCwalina**
We cannot have multiple types called e.g. Utf8, even if they are in different namespaces. Our ecosystem of tools gets confused with such APIs). This moves Utf8, Utf16, and Custom encoders, parsers, formatters to be nested types in Encodings, Parsers, Formatters outer classes.

Also, I moved all the APIs into System.Buffers* namesapces in preparation for the move to System.Buffers.dll

cc: @ahsonkhan, @shiftylogic, @joshfree

### CODE REVIEW

File: src/System.Text.Primitives/System/Text/Extensions.Integer.cs [hunk 1]
```diff
@@ -1,7 +1,9 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-namespace System.Text
+using System.Buffers;
+
+namespace System.Buffers.Text
```
Comment (ahsonkhan) [inline src/System.Text.Primitives/System/Text/Extensions.Integer.cs]: Why are the TryFormat extension methods in the System.Buffers.Text namespace when the Custom Formatter TryFormat are in System.Buffers namespace? Should the Extensions class also be in System.Buffers namespace?

**Comment by ahsonkhan** [comment]
> Also, I moved all the APIs into System.Buffers* namesapces in preparation for the move to System.Buffers.dll


Can you summarize what is in the System.Buffers.Text namespace?

**Comment by KrzysztofCwalina** [comment]
I left Extensions and SymbolTable in System.Buffers.Text.
I think the Extensions class should simply go away. It does not add much value above the Formatters APIs. It just defaults some parameters, which are mostly not needed in the Utf8 and Utf16 inner class APIs. See https://github.com/dotnet/corefxlab/blob/master/src/System.Text.Primitives/System/Text/Extensions.Float.cs

SymbolTable could be moved somewhere else, but it's so specific to Text. Having said that I don't feel strongly either way.
